### ISSUE TITLE
Inconsistent highlighting of errors.

### ISSUE THREAD

**Issue by RikardLegge** [SublimeLinter/SublimeLinter-flow#13]
I've found the error highlighting quite inconsistent. 

Examples:
#### test_main.js:
```javascript
/* @flow */

// FAIL: requried an invalid path. (Only the first letter is highlighted)
var require_invalid = require('./test_invalid.js');
// SUCCESS: require a valid path
var test_required = require('./test_required.js');

// FAIL: Multiplying a string with a number, not valid in javascript. (The whole line gets maked as an error, not only the error itself)
var thisIsAnError = a*b; // Also highlighted

// Doesn't get highlighted, since the values originate from test_reguested.js.
thisIsAnError = test_required.num*test_required.str;

// SUCCESS: A simple function with a non optional string and an optionan number as it's input. 
function lengthOfString(name:string, someNumber?:number){
	someNumber = someNumber || 5;
	return name + "'s length is " + name.length + " & " + someNumber;
}

// FAIL: invalid input. (Only the first letter is highlighted)
size("user","te");

// Error: can't multiply a string an a number. Correctly highlighted.
var str = "a string" * 2;
```
#### test_requested.js
```javascript
/* @flow */

// For testing in main file. 
module.exports.num = 1;

// For testing in main file. Displays an error even though the error originated from test_main.js and should be highlighted there.
module.exports.str = "a string";
```

#### Flow output
```
.../test_main.js:4:23,50: ./test_invalid.js
Required module not found

.../test_main.js:9:21,21: identifier a
Unknown global name

.../test_main.js:9:23,23: identifier b
Unknown global name

.../test_main.js:21:1,4: identifier size
Unknown global name

.../test_main.js:24:11,20: string
This type is incompatible with
.../test_main.js:24:11,24: number

.../test_required.js:7:22,31: string
This type is incompatible with
  .../test_main.js:12:16,50: number
```

**Comment by RikardLegge** [comment]
See #9 regarding related problems and proposed solution.

**Comment by ckaznocha** [comment]
Closed by PR #21
